apm-sniffer/apm-sdk-plugin/spymemcached-2.x-plugin/src/main/java/org/skywalking/apm/plugin/spymemcached/v2/MemcachedMethodInterceptor.java

```diff
@@ -0,0 +1,47 @@
+package org.skywalking.apm.plugin.spymemcached.v2;
+
+import java.lang.reflect.Method;
+
+import org.skywalking.apm.agent.core.context.ContextManager;
+import org.skywalking.apm.agent.core.context.tag.Tags;
+import org.skywalking.apm.agent.core.context.trace.AbstractSpan;
+import org.skywalking.apm.agent.core.context.trace.SpanLayer;
+import org.skywalking.apm.agent.core.plugin.interceptor.enhance.EnhancedInstance;
+import org.skywalking.apm.agent.core.plugin.interceptor.enhance.InstanceMethodsAroundInterceptor;
+import org.skywalking.apm.agent.core.plugin.interceptor.enhance.MethodInterceptResult;
+import org.skywalking.apm.logging.ILog;
+import org.skywalking.apm.logging.LogManager;
+import org.skywalking.apm.network.trace.component.ComponentsDefine;
+
+public class MemcachedMethodInterceptor implements InstanceMethodsAroundInterceptor {
+
+    private static final ILog logger = LogManager.getLogger(MemcachedMethodInterceptor.class);
+    
+    @Override 
+    public void beforeMethod(EnhancedInstance objInst, Method method, Object[] allArguments,
+        Class<?>[] argumentsTypes, MethodInterceptResult result) throws Throwable {
+        Object[] arguments = allArguments;
+        String peer = String.valueOf(objInst.getSkyWalkingDynamicField());
+        AbstractSpan span = ContextManager.createExitSpan("SpyMemcached/" + method.getName(), peer);
+        span.setComponent(ComponentsDefine.MEMCACHE);
+        Tags.DB_TYPE.set(span, "Memcache");
```

Comment:
`Memcache` should be a constant.

Reply:
OK, I write as redis plugin : Tags.DB_TYPE.set(span, "Redis");

Reply:
I think that is a terrible sample code, Need to fix, @ascrutae

Reply:
I will fix it. @wu-sheng